exchanges/binanceus/type_convert.go

```diff
@@ -0,0 +1,617 @@
+	a.Data.EventTime = aux.Data.EventTime.Time()
+	a.Data.LastUpdated = aux.Data.LastUpdated.Time()
+	return nil
+}
+
+// UnmarshalJSON deserialises the JSON info, including the (Timestamp)timestamp
+func (a *WebsocketDepthStream) UnmarshalJSON(data []byte) error {
+	type Alias WebsocketDepthStream
+	aux := &struct {
+		Timestamp binanceusTime `json:"E"`
+		*Alias
+	}{
+		Alias: (*Alias)(a),
+	}
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+	a.Timestamp = aux.Timestamp.Time()
+	return nil
+}
+
+// UnmarshalJSON .. .
+func (a *WebsocketAggregateTradeStream) UnmarshalJSON(data []byte) error {
+	type Alias WebsocketAggregateTradeStream
+	chil := &struct {
+		*Alias
+		TradeTime int64 `json:"T"`
+		EventTime int64 `json:"E"`
+	}{
+		Alias: (*Alias)(a),
+	}
+	if err := json.Unmarshal(data, chil); err != nil {
+		return err
+	}
+	if chil.TradeTime > 0 {
+		a.TradeTime = time.UnixMilli(chil.TradeTime)
+	}
+	if chil.EventTime > 0 {
+		a.EventTime = time.UnixMilli(chil.EventTime)
+	}
+	return nil
+}
+
+// binanceTime provides an internal conversion helper
+type binanceusTime time.Time
+
+func (t *binanceusTime) UnmarshalJSON(data []byte) error {
+	var timestamp int64
+	if err := json.Unmarshal(data, &timestamp); err != nil {
+		return err
+	}
+	*t = binanceusTime(time.UnixMilli(timestamp))
+	return nil
+}
+
+// Time returns a time.Time object
+func (t binanceusTime) Time() time.Time {
+	return time.Time(t)
+}
+
+// UnmarshalJSON deserialises createTime timestamp to built in time.
+func (a *OCBSOrder) UnmarshalJSON(data []byte) error {
+	type Alias OCBSOrder
+	chil := &struct {
+		*Alias
+		CreateTime int64 `json:"createTime"`
+	}{
+		Alias: (*Alias)(a),
+	}
+	if er := json.Unmarshal(data, chil); er != nil {
+		return er
+	}
+	if chil.CreateTime > 0 {
+		a.CreateTime = time.UnixMilli(chil.CreateTime)
+	}
+	return nil
+}
+
+// UnmarshalJSON deserialises createTime timestamp to built in time.
+func (a *ServerTime) UnmarshalJSON(data []byte) error {
+	type Alias ServerTime
+	chil := &struct {
+		*Alias
+		Timestamp int64 `json:"serverTime"`
+	}{
+		Alias: (*Alias)(a),
+	}
+	if er := json.Unmarshal(data, chil); er != nil {
+		return er
+	}
+	if chil.Timestamp > 0 {
+		a.Timestamp = time.UnixMilli(chil.Timestamp)
+	}
+	return nil
+}
+
+// UnmarshalJSON deserialises createTime timestamp to built in time.
+func (a *SubAccountStatus) UnmarshalJSON(data []byte) error {
+	type Alias SubAccountStatus
+	chil := &struct {
+		*Alias
+		InsertTime int64 `json:"insertTime"`
+	}{
+		Alias: (*Alias)(a),
+	}
+	if er := json.Unmarshal(data, chil); er != nil {
+		return er
+	}
+	if chil.InsertTime > 0 {
+		a.InsertTime = time.UnixMilli(chil.InsertTime)
+	}
+	return nil
+}
+
+// UnmarshalJSON deserialises ValidTimestamp timestamp to built in time.Time instance.
+func (a *Quote) UnmarshalJSON(data []byte) error {
+	type Alias Quote
+	chil := &struct {
+		*Alias
+		ValidTimestamp int64 `json:"validTimestamp,string"`
```

Comment:
`{"symbol":"BTCUSDT","ratio":"23381","inverseRatio":"0.00004277","validTimestamp":1659321576885,"toAmount":"23381","fromAmount":"1"}`
this is not a string